synapseclient/models/team.py

```diff
@@ -1 +1,304 @@
-# TODO
+import asyncio
+from dataclasses import dataclass
+from typing import Optional, Dict, List
+from opentelemetry import trace, context
+
+from synapseclient import Synapse
+from synapseclient.team import (
+    Team as Synapse_Team,
+    TeamMember as Synapse_TeamMember,
+)
+from synapseclient.models.user import UserGroupHeader
+from synapseclient.core.async_utils import otel_trace_method
+
+
+tracer = trace.get_tracer("synapseclient")
+
+
+@dataclass
+class TeamMember:
+    """
+    Contains information about a user's membership in a Team.
+    In practice the constructor is not called directly by the client.
+
+    Attributes:
+        team_id: The ID of the team
+        member: An object of type [org.sagebionetworks.repo.model.UserGroupHeader](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/UserGroupHeader.html)
+                describing the member
+        is_admin: Whether the given member is an administrator of the team
+    """
+
+    team_id: Optional[int] = None
+    """The ID of the team"""
+
+    member: Optional[UserGroupHeader] = None
+    """An object of type [org.sagebionetworks.repo.model.UserGroupHeader](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/UserGroupHeader.html)"""
+
+    is_admin: Optional[bool] = None
+    """Whether the given member is an administrator of the team"""
+
+    def fill_from_dict(self, synapse_team_member: Synapse_TeamMember) -> "TeamMember":
```

Comment:
I noted in Bryan's PR too about the type hint for `synapse_team_member`.  Should it contain `dict` as well?